lib/chef/application.rb

```diff
@@ -198,16 +204,19 @@ def run_application
 
     # Initializes Chef::Client instance and runs it
     def run_chef_client(specific_recipes = [])
+      unless specific_recipes.respond_to?(:size)
```

Comment:
@tyler-ball I had a fatal error here and it broke tons of other tests, so switched the behavior out to warn & spec'd that.

as for `#kind_of?`, that should be avoided, I think, since it explicitly binds this code to Arrays -- one should check if a given object smells as though it can be used in a particular way (check the objects interfaces). This way, subclasses of array, or anything that implements `#size`, can be used here.

I'm indifferent though and just want this fixed. If you would prefer that, I can make it so. The test will have to be updated as well, and this will uncover a number of other (>30 IIRC) failures.